**Assignment on Measures of Central Tendency and Dispersion**

An Institution wishes to find out their student’s ability in mathematics, reading
and writing skills. The Institution wants to do an exploratory study to check the
following information.

**1. Find out how many males and females participated in the test.**

In [1]:
#Firstly we need to import the necessary libraries
import pandas as pd
import numpy as np

In [2]:
#Loading the dataset
stperform=pd.read_csv('/content/StudentsPerformance.csv')

In [3]:
stperform.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33

In [ ]:
stperform.columns

Index(['gender', 'race/ethnicity', 'parental level of education', 'lunch',
       'test preparation course', 'math score', 'reading score',
       'writing score'],
      dtype='object')

In [ ]:
#To find the number of males and females
stperform.gender.value_counts()

female    518
male      482
Name: gender, dtype: int64

There are 518 females and 482 males participated in the test

**2. What do you think about the students' parental level of education?**

In [ ]:
stperform['parental level of education'].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

Most of the students parents has gone to some college and the least number of students parents has a master's degree

**3. Who scores the most on average for math, reading and writing based on**

**Gender**

In [19]:
data1=stperform.groupby('gender')

In [20]:
data1[['math score','reading score','writing score']].mean()

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

The average math score of males is more than that of females. In the case of reading score the average score of females is more than that of males. For writing score the average score of females is more than that of males.  

**Test preparation course**

In [18]:
data2=stperform.groupby('test preparation course')

In [21]:
data2[['math score','reading score','writing score']].mean()

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

The average 'math score','reading score' and 'writing score' of people completed test preparation course is more than that of those who didn't complete it.

**4. What do you think about the scoring variation for math, reading and writing
based on-**

**● Gender**

In [22]:
data1[['math score','reading score','writing score']].std()

math score  reading score  writing score
gender                                          
female   16.029928      14.411018      14.844842
male     14.556411      14.149594      14.227225

Based on gender the variation in math, reading and writing score for female is more than that of variation for the scores of male.   

**● Test preparation course**

In [23]:
data2[['math score','reading score','writing score']].std()

math score  reading score  writing score
test preparation course                                          
completed                 14.521847      13.537572      13.236412
none                      15.705689      14.608896      15.041667

Based on test preparation course the variation in math, reading and writing scores for students who didn't complete course is more than that of variation of the scores of students who completed the course.

**5. The management needs your help to give bonus points to the top 25% of
students based on their math score, so how will you help the management
to achieve this.**

In [ ]:
#First find the 3rd quartile value
np.quantile(stperform['math score'], 0.75)

77.0

In [ ]:
#Based on the 3rd quartile value top 25% observations lie above the math mark 77 we need to find these observations
stperform[stperform['math score']>77]
#The given below students are the top 25% based on their math score

gender race/ethnicity parental level of education         lunch  \
430  female        group A                some college      standard   
431  female        group B           bachelor's degree  free/reduced   
432  female        group D           bachelor's degree      standard   
433  female        group D                 high school      standard   
434  female        group D           bachelor's degree  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
430               completed          78             87             91  
431                    none          78             79             76  
432                    none          78             82             79  
433                    none          78             81             80  
434                    none          78             90             93  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[231 rows x 8 columns]